crates/engine/tree/src/tree/mod.rs

```diff
@@ -2294,7 +2303,20 @@ where
         // Extend with block we are validating root for.
         input.append_ref(hashed_state);
 
-        Ok(ParallelStateRoot::new(consistent_view, input).incremental_root_with_updates()?)
+        let (tx, rx) = std::sync::mpsc::channel();
+        let blocking_task_pool = self.blocking_task_pool.clone();
+
+        rayon::spawn(move || {
+            let result: Result<(B256, TrieUpdates), AsyncStateRootError> =
+                futures::executor::block_on(async move {
+                    AsyncStateRoot::new(consistent_view, blocking_task_pool, input)
+                        .incremental_root_with_updates()
+                        .await
+                });
```

Comment:
this seems like a very bad idea, because this blocks the current thread, rayon really does not like this. this is what I meant with the async state root impl doesn't really have to async because it only uses async since it uses tokio channels.
which we now await in a blocking manner